test/statements/serial/serialReturn.chpl

```diff
@@ -0,0 +1,21 @@
+use Time;
+
+proc serialWithReturn() {
+    serial do return 1;
+}
+
+proc test() {
+
+  serialWithReturn();
+
+  begin{
+    sleep(1);
```

Comment:
Hi @rahulghangas: Thanks for adding the test.  I feel nervous about having it rely on a `sleep()` because this has gotten us into trouble in the past since there's nothing, semantically speaking, that guarantees that the second task will run while the first is sleeping (even though typically it will be the case).  At times, such assumptions have caught us in testing every so often due to different tasking or timing behaviors.

Could we change this to some sort of sync variable latch between the two tasks such that in the old/buggy version, we'd essentially get a deadlock (because the first task would be stuck waiting for the second to let it proceed) but the fix would make it work?  I think the following would do it:

```chapel
use Time;

proc serialWithReturn() {
    serial do return 1;
}

proc test() {
  var x$: sync int;
  
  serialWithReturn();

  begin{
    x$; // read flag
    writeln("Begin 1");
  }

  begin{
    writeln("Begin 2");
    x$ = 1;
  }
}

test();
```

?  (but definitely double-check my thinking)

Reply:
Yep, this works. I was thinking if someone does break it, the test would simply stay idle instead of returning a bad value. Is that okay?

Reply:
Our testing has timeouts that should kill the job if this occurs, so I think it's OK.  Also, if someone breaks it, they should find out during their development cycle rather than in our nightly testing, so the developer will suffer, not the nightly results.

Reply:
sounds good, I'll update the PR